lib/zipkin-tracer/trace.rb

```diff
@@ -56,10 +46,32 @@ def to_h
       }
     end
 
+    # We record information into spans, then we send these spans to zipkin
+    def record(value, endpoint = Trace.default_endpoint)
+      stack(Trace::Annotation.new(value, endpoint))
+    end
+
+    def record_tag(key, value, type = Trace::BinaryAnnotation::Type::STRING, endpoint = Trace.default_endpoint)
+      stack(Trace::BinaryAnnotation.new(key, value, type, endpoint))
+    end
+
+    def record_local_component(value)
+      record_tag(BinaryAnnotation::LOCAL_COMPONENT, value)
+    end
+
     private
 
     UNKNOWN_DURATION = 0 # mark duration was not set
 
+    def stack(annotation)
+      case annotation
+      when BinaryAnnotation
+        binary_annotations << annotation
+      when Annotation
+        annotations << annotation
+      end
+    end
```

Comment:
IMHO now the methods above are clearly separate I don't think we need to have this abstraction, we can add to the respective arrays directly in the methods, e.g. `annotations << Trace::Annotation.new(value, endpoint)` in `record` and `binary_annotations << Trace::BinaryAnnotation.new(key, value, type, endpoint)` in `record_tag`